dts/bindings/input/zephyr,gpio-qdec.yaml

```diff
@@ -0,0 +1,63 @@
+# Copyright 2023 Google LLC
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  GPIO based QDEC input device
+
+  Implement an input device generating relative axis event reports for a rotary
+  encoder connected to two GPIOs. The driver is normally idling until it sees a
+  transition on any of the encoder signal lines, then switches to polling mode
+  and samples the two signal lines periodically to track the encoder position,
+  and goes back to idling after the specified timeout.
+
+  Example configuration:
+
+  #include <dt-bindings/input/input-event-codes.h>
+
+  qdec {
+          compatible = "zephyr,gpio-qdec";
+          gpios = <&gpio0 14 (GPIO_PULL_UP | GPIO_ACTIVE_HIGH)>,
+                  <&gpio0 13 (GPIO_PULL_UP | GPIO_ACTIVE_HIGH)>;
+          steps-per-period = <4>;
+          zephyr,axis = <INPUT_REL_Y>;
+          sample-time-us = <2000>;
+          idle-timeout-ms = <200>;
+  };
+
+compatible: "zephyr,gpio-qdec"
```

Comment:
I wonder if the `zephyr,` prefix is really needed, this is a generic peripheral, like `gpio-keys`.

Reply:
I really have no preference but for `gpio-keys` we have both `gpio-keys` for the binding alone and `zephyr,gpio-keys` for the binding associated with the driver. Not too sure why we went that way, on regulator we have `regulator-fixed` but then there's plenty other `zephyr,...` bindings defined. Not sure what to do here.

Reply:
@galak ? I think that bindings representing generic hardware, like regulator-fixed, regulator-gpio, gpio-keys, etc do not have any prefix. In the end there's nothing Zephyr specific in there (except input codes?)

Reply:
@galak ping

Reply:
-EGALAKTIMEOUT, dropped the prefix